src/exoticatechnologies/modifications/exotics/impl/GuardianShield.kt

```diff
@@ -120,9 +125,91 @@ class GuardianShield(key: String, settings: JSONObject) : Exotic(key, settings)
+     *
+     * Defaults to (1 - [HARDFLUX_DEBUFF_AMOUNT])
+     */
+    private fun getHardFluxDebuffAmount(member: FleetMemberAPI, mods: ShipModifications, exoticData: ExoticData): Float {
+        return (1f - (HARDFLUX_DEBUFF_AMOUNT * getNegativeMult(member, mods, exoticData))).coerceIn(0f, 1f)
+    }
+
+    /**
+     * Method for calculating the radius, or rather how big the active Guardian Shield radius will be, after
+     * applying the [getPositiveMult] modifier to it.
+     *
+     * Defaults to [ACTIVE_SHIELD_RADIUS]
+     */
+    private fun getRadiusAmount(member: FleetMemberAPI, mods: ShipModifications, exoticData: ExoticData): Float {
+        return ACTIVE_SHIELD_RADIUS * getPositiveMult(member, mods, exoticData)
+    }
+
+    /**
+     * Method for calculating the shield expansion factor, or rather how good the hard flux regeneration will be, after
+     * applying the [getPositiveMult] modifier to it.
+     *
+     * Defaults to [SHIELD_EXPANDING_FACTOR]
+     */
+    private fun getShieldExpansionAmount(member: FleetMemberAPI, mods: ShipModifications, exoticData: ExoticData): Float {
+        return SHIELD_EXPANDING_FACTOR * getPositiveMult(member, mods, exoticData)
+    }
+
+    /**
+     * Method for calculating the shield push out effect factor, or rather how strong the shield will push out enemies,
+     * after applying the [getPositiveMult] modifier to it
+     *
+     * Defaults to:
+     * - 0 for null
+     * - 1 for [HullSize.DEFAULT]
+     * - 1 for [HullSize.FIGHTER]
+     * - 2 for [HullSize.FRIGATE]
+     * - 2.5 for [HullSize.DESTROYER]
+     * - 3.5 for [HullSize.CRUISER]
+     * - 5 for [HullSize.CAPITAL_SHIP]
+     */
+    private fun getShieldPushOutEffectMomentumFactor(
+            hullSize: HullSize?,
+            member: FleetMemberAPI,
+            mods: ShipModifications,
+            exoticData: ExoticData
+    ): Float {
+        return when (hullSize) {
+            null -> 0f
+            HullSize.DEFAULT -> 1f
+            HullSize.FIGHTER -> 1f
+            HullSize.FRIGATE -> 2f
+            HullSize.DESTROYER -> 2.5f
+            HullSize.CRUISER -> 3.5f
+            HullSize.CAPITAL_SHIP -> 5f
+        }.exhaustive * getPositiveMult(member, mods, exoticData)
+    }
+
+    private fun generateFluxEffectString(member: FleetMemberAPI, mods: ShipModifications, exoticData: ExoticData): String {
+        return lazyFluxEffectString
+                .replace(FLUX_DEBUFF_TOKEN, Math.round(getFluxDebuffAmount(member, mods, exoticData) * 100f).toString())
+                .replace(HARDFLUX_DEBUFF_TOKEN, Math.round(getHardFluxDebuffAmount(member, mods, exoticData) * 100f).toString())
+    }
+
+    inner class GuardianShieldDrone(
+            ship: ShipAPI,
+            val member: FleetMemberAPI,
+            val mods: ShipModifications,
+            val exoticData: ExoticData
+    ) : MagicDroneSubsystem(ship) {
 
         var engine: CombatEngineAPI = Global.getCombatEngine()
+        @Volatile
```

Comment:
i think the whole synchronization is very unnecessary as the problem was rooted somewhere else but meh, it can stay now that time was ~~wasted~~ spent on it and it works alright.

Reply:
i'll let it unnecessarily stay since it's working alright. resolving.